src/controllers/authenticationController.ts

```diff
@@ -449,7 +497,23 @@ export default class AuthenticationController {
       user.session_token
     );
 
-    return ResponsePermanentRedirect(urlWithToken);
+    return pipe(
+      isUserElegibleForIoLoginUrlScheme(user.fiscal_code),
+      B.fold(
+        () => urlWithToken,
+        () =>
+          pipe(
+            clientProfileRedirectionUrl,
+            S.replace("{token}", user.session_token),
+            S.replace(defaultUrlSchemeRegex, "iologin:"),
+            UrlFromString.decode,
+            E.getOrElseW((err) => {
+              throw new Error(`Invalid url | ${readableReportSimplified(err)}`);
+            })
+          )
```

Comment:
ditto. We could reuse `urlWithToken` and replace the `defaultUrlSchemeRegex`.

Also, we can try to extract the replacement into a function and use it in both line 184 and line 505.

Reply:
refactored in https://github.com/pagopa/io-backend/pull/1012/commits/d28fdcf11cd794984c55d974a73f22bb27ffb71b